includes/class-wc-webhook.php

```diff
@@ -349,7 +351,7 @@ private function build_payload( $resource_id ) {
 				'id' => $resource_id,
 			);
 		} else {
-			if ( 'wp_api_v1' === $this->get_api_version() ) {
+			if ( in_array( $this->get_api_version(), array( 'wp_api_v1', 'wp_api_v2', true ) ) ) {
```

Comment:
@mikejolley / @claudiosanches this line introduces `true` which means the in array will pass even if the `$this->get_api_version()` returns `legacy_v3`